pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -236,9 +258,18 @@ public String segmentCommit(@QueryParam(SegmentCompletionProtocol.PARAM_INSTANCE
     SegmentCompletionProtocol.Response response = segmentCompletionManager.segmentCommitStart(requestParams);
     if (response.equals(SegmentCompletionProtocol.RESP_COMMIT_CONTINUE)) {
       // Get the segment and put it in the right place.
-      boolean success = uploadSegment(multiPart, instanceId, segmentName, false) != null;
+      ImmutablePair<URI, SegmentMetadataImpl>  uploadResult = uploadSegment(multiPart, instanceId, segmentName, false);
+      boolean success =  (uploadResult != null && uploadResult.getLeft() != null && uploadResult.getRight() != null);
+      SegmentMetadataImpl segmentMetadata = null;
```

Comment:
we can fail the commit if segment metadata is null.

Reply:
Done.